src/config.rs

```diff
@@ -409,6 +493,10 @@ impl Config {
         Ok(())
     }
 
+    pub fn get_bad_node_informer(&self) -> Option<BadNodeInformer> {
```

Comment:
FYI You don't need to call this `get_`, Rust has UFCS, so there's no way for `self.bad_node_informer` to be confused as a function, as `self.bad_node_informer()` is sugar for `Config::bad_node_informer(self)`.
```suggestion
    pub fn bad_node_informer(&self) -> Option<BadNodeInformer> {
```